tests/testthat/test-driver-snowflake.R

```diff
@@ -189,3 +194,689 @@ test_that("Workbench-managed credentials are ignored for other accounts", {
+
+  expect_equal(config$default_connection_name, "prod")
+  expect_equal(config$connections$dev$account, "devaccount")
+})
+
+test_that("load_snowflake_config() merges connections.toml over config.toml", {
+  skip_if_not_installed("toml")
+
+  config_dir <- tempfile()
+  dir.create(config_dir)
+  withr::defer(unlink(config_dir, recursive = TRUE))
+
+  writeLines(
+    c(
+      'default_connection_name = "prod"',
+      '',
+      '[connections.dev]',
+      'account = "from_config"',
+      'user = "config_user"'
+    ),
+    file.path(config_dir, "config.toml")
+  )
+
+  writeLines(
+    c(
+      '[dev]',
+      'account = "from_connections"',
+      'user = "connections_user"',
+      'warehouse = "DEV_WH"'
+    ),
+    file.path(config_dir, "connections.toml")
+  )
+
+  withr::local_envvar(SF_SKIP_WARNING_FOR_READ_PERMISSIONS_ON_CONFIG_FILE = "true")
+
+  config <- load_snowflake_config(config_dir)
+
+  # connections.toml completely overwrites
+  expect_equal(config$default_connection_name, "prod")  # From config.toml
+  expect_equal(config$connections$dev$account, "from_connections")
+  expect_equal(config$connections$dev$warehouse, "DEV_WH")
+})
+
+test_that("load_snowflake_config() handles missing files gracefully", {
+  skip_if_not_installed("toml")
+
+  config_dir <- tempfile()
+  dir.create(config_dir)
+  withr::defer(unlink(config_dir, recursive = TRUE))
+
+  config <- load_snowflake_config(config_dir)
+
+  expect_equal(config$default_connection_name, "default")
+  expect_equal(length(config$connections), 0)
+})
+
+test_that("SNOWFLAKE_CONNECTIONS env var overrides file-based connections", {
+  skip_if_not_installed("toml")
+
+  withr::local_envvar(
+    SNOWFLAKE_CONNECTIONS = '[prod]
+account = "envaccount"
+user = "envuser"',
+    SF_SKIP_WARNING_FOR_READ_PERMISSIONS_ON_CONFIG_FILE = "true"
+  )
+
+  config_dir <- tempfile()
+  dir.create(config_dir)
+  withr::defer(unlink(config_dir, recursive = TRUE))
+
+  writeLines(
+    c('[prod]', 'account = "fileaccount"'),
+    file.path(config_dir, "connections.toml")
+  )
+
+  config <- load_snowflake_config(config_dir)
+
+  expect_equal(config$connections$prod$account, "envaccount")
+})
+
+test_that("SNOWFLAKE_DEFAULT_CONNECTION_NAME env var overrides config file", {
+  skip_if_not_installed("toml")
+
+  withr::local_envvar(
+    SNOWFLAKE_DEFAULT_CONNECTION_NAME = "staging",
+    SF_SKIP_WARNING_FOR_READ_PERMISSIONS_ON_CONFIG_FILE = "true"
+  )
+
+  config_dir <- tempfile()
+  dir.create(config_dir)
+  withr::defer(unlink(config_dir, recursive = TRUE))
+
+  writeLines(
+    'default_connection_name = "prod"',
+    file.path(config_dir, "config.toml")
+  )
+
+  config <- load_snowflake_config(config_dir)
+
+  expect_equal(config$default_connection_name, "staging")
+})
+
+test_that("Case 1: Named connection loads and merges with kwargs", {
+  skip_if_not_installed("toml")
+
+  config_dir <- tempfile()
+  dir.create(config_dir)
+  withr::defer(unlink(config_dir, recursive = TRUE))
```

Comment:
```suggestion
  config_dir <- withr::local_tempdir()
```

And elsewhere in this test file